src/Microsoft.Graph.Core/Requests/Content/BatchResponseContent.cs

```diff
@@ -0,0 +1,146 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Text;
+    using System.Threading.Tasks;
+    /// <summary>
+    /// Handles batch request responses.
+    /// </summary>
+    public class BatchResponseContent
+    {
+        private JObject jBatchResponseObject;
+        private HttpResponseMessage batchResponseMessage;
+
+        /// <summary>
+        /// Constructs a new <see cref="BatchResponseContent"/>
+        /// </summary>
+        /// <param name="httpResponseMessage">A <see cref="HttpResponseMessage"/> of a batch request execution.</param>
+        public BatchResponseContent(HttpResponseMessage httpResponseMessage)
+        {
+            this.batchResponseMessage = httpResponseMessage ?? throw new ServiceException(new Error
+            {
+                Code = ErrorConstants.Codes.InvalidRequest,
+                Message = string.Format(ErrorConstants.Messages.NullParameter, "httpResponseMessage")
+            });
+        }
+
+        /// <summary>
+        /// Gets all batch responses <see cref="Dictionary{String, HttpResponseMessage}"/>.
+        /// </summary>
+        /// <returns>A Dictionary of id and <see cref="HttpResponseMessage"/> representing batch responses.</returns>
+        public async Task<Dictionary<string, HttpResponseMessage>> GetResponsesAsync()
```

Comment:
As httpresponsemessage is Idisposable and you're allocating a bunch of those I'd probably add a wanring in the method documentation to tell people to dispose.